module/zfs/dsl_prop.c

```diff
@@ -69,6 +70,16 @@ dodefault(zfs_prop_t prop, int intsz, int numints, void *buf)
 	return (0);
 }
 
+static int
+dsl_prop_known_index(zfs_prop_t prop, uint64_t value)
+{
+	const char *str = NULL;
+	if (zfs_prop_get_type(prop) == PROP_TYPE_INDEX)
+		return (!zfs_prop_index_to_string(prop, value, &str));
+
+	return (-1);
```

Comment:
We typically use return value of `0` for success, and positive errno's for errors.  so I would use `return (EINVAL)` here.  Or it might be simpler to return `boolean_t` since really we are asking, "is this known?  yes/no"

Reply:
Here `dsl_prop_known_index` returns 1 if the property index is known, returns 0 if the property index is unknown, and returns -1 if it is not an index-type property. @behlendorf has mentioned [here](https://github.com/openzfs/zfs/pull/13680#discussion_r966268129) this may be useful in the future.